### ISSUE TITLE
`Failed to get D-Bus connection: No such file or directory` when trying to deploy

### ISSUE THREAD

**Issue by jmuheim** [seuros/capistrano-puma#348]
When trying to deploy I get a `Failed to get D-Bus connection: No such file or directory`:

![CleanShot 2022-10-06 at 08 53 51@2x](https://user-images.githubusercontent.com/1814983/194234321-204c8e1c-8cd0-45fd-8ec6-1f41f2509d7c.png)

Any help is highly appreciated.

**Comment by seuros** [comment]
What OS and version is this ?

**Comment by jmuheim** [comment]
I'm on macOS. And it was the most current version of the gem.

My server is a linux one, do you need more details?

**Comment by seuros** [comment]
I need server details . Version and flavor.

I testing in debian and ubuntu. 20.04+

**Comment by jmuheim** [comment]
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

Does that help?

**Comment by jmuheim** [comment]
No need to hurry, I’ll be able to deploy only next Monday again anyway. 🙂

**Comment by seuros** [comment]
https://github.com/seuros/capistrano-puma/blob/50d9b7e027193a8b6329c59c77add7005cab4485/lib/capistrano/puma/systemd.rb#L14

You should set this variable to the systemctl in centos.

Use master branch for now.

**Comment by seuros** [comment]
> No need to hurry, I’ll be able to deploy only next Monday again anyway. 🙂


Cool. I will have it tested by then.

**Comment by seuros** [comment]
I noticed that Centos7 has pretty old config.  You will need to add the deploy user to some groups to access the dbus. 

or 

you can use
```
set :systemctl_user, :system
```

To use the service with sudo.  Your deploy user should be able to `sudo`

**Comment by c-project** [comment]
The same problem on my env after update.
Previously I added all command to sudoers to be able to run any command under `deploy` user. And it worked perfectly except one cap command (that I fixed in my PR)
Now it runs everything with `--user` that raises error mentioned in issue.
If I run all under system user it adds `sudo` explicitly what require to enter sudo password.
```
00:00 puma:status
      01 sudo /bin/systemctl status app_puma_production
      01 [sudo] password for deploy:
```
